actionpack/lib/action_controller/metal/rack_delegation.rb

```diff
@@ -8,9 +8,15 @@ module RackDelegation
     delegate :headers, :status=, :location=, :content_type=,
              :status, :location, :content_type, :response_code, :to => "@_response"
 
-    def dispatch(action, request)
+    module ClassMethods
```

Comment:
Right. The place now make sense but the name is still not ideal. Maybe `controller_for_env`?